pints/_mcmc/_adaptive_covariance_localised.py

```diff
@@ -0,0 +1,245 @@
+        self._update_mu()
+        self._update_sigma()
+        self._update_w()
+        self._update_lambda()
+        self._update_alpha()
+        
+    def _update_log_q(self):
+        """
+        Updates log q values representing weights of Gaussian mixture
+        components. If first time this is called, then
+        this function creates q functions
+        """
+        self._log_q_l = np.zeros(self._mixture_components)
+        for i in range(self._mixture_components):
+            self._log_q_l[i] = (np.log(self._w[i]) +
+                                   scipy.stats.multivariate_normal.logpdf(self._current,
+                                                         self._mu[i], self._sigma[i],
+                                                         allow_singular=True))
+        self._log_q_l += -logsumexp(self._log_q_l)
+
+    def _update_mu(self):
+        """
+        Updates mu components according to,
+        mu_t+1^k = mu_t^k + gamma_t+1 * q_t^k * (theta_t+1 - mu_t^k)
+        """
+        for i in range(self._mixture_components):
+            self._mu[i]  = ((1 - self._gamma * np.exp(self._log_q_l[i])) * self._mu[i] +
+                            self._gamma * np.exp(self._log_q_l[i]) * self._current)
+
+    def _update_sigma(self):
+        """
+        Updates sigma components according to,
+        sigma_t+1^k = sigma_t^k + gamma_t+1 * q_t^k *
+                      ((theta_t+1 - mu_t^k)(theta_t+1 - mu_t^k)' - sigma_t^k)
+        """
+        for i in range(self._mixture_components):
+            dsigm = np.reshape(self._current - self._mu[i], (self._dimension, 1))
+            self._sigma[i] = self._sigma[i] + self._gamma * np.exp(self._log_q_l[i]) * (
+                             np.dot(dsigm, dsigm.T) - self._sigma[i])
+
+    def _update_w(self):
+        """
+        Updates w components according to,
+        w_t+1^k = w_t^k + gamma_t+1 * (q_t^k - w_t^k)
+        """
+        for i in range(self._mixture_components):
+            self._w[i] += self._gamma * (np.exp(self._log_q_l[i]) - self._w[i])
+
+    def _update_lambda(self):
+        """
+        Updates lambda components according to,
+        log lambda_t+1^k = log lambda_t^k + gamma_t+1 * 1(Z_t+1==k?) *
+                             (alpha_k(theta_t, Y_t+1) - self._target_acceptance)
+        """
+        # Only update Zth component
+        self._log_lambda[self._Z] += (self._gamma *
+                                     (self._alpha - self._target_acceptance))
+      
+    def _update_alpha(self):
+        """
+        Updates running acceptance probabilities according to,
+        alpha_t+1^k = alpha_t^k + gamma_t+1 * 1(Z_t+1==k?) * 
+                             (alpha_k(theta_t, Y_t+1) - alpha_t^k)
+        """
+        self._alpha_l[self._Z] += (self._gamma *
+                                     (self._alpha - self._alpha_l[self._Z]))
+
+    def _ratio_q(self):
+        """
+        Yields log q(Y_t+1|Z_t+1) - log q(theta_t|Z_t+1)
```

Comment:
Returns, not yields! (`yield` is a different way of returning in Python, used to create iterators)